sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/document_producer.py

```diff
@@ -271,3 +271,51 @@ def _validate_orderby_items(self, res1, res2):
             type2 = _OrderByHelper.getTypeStr(elt2)
             if type1 != type2:
                 raise ValueError("Expected {}, but got {}.".format(type1, type2))
+
+class _NonStreamingDocumentProducer(object):
+    """This class takes care of handling of the items to be sorted in a non-streaming context.
+    One instance of this document producer goes attached to every item coming in for the priority queue to be able
+    to properly sort items as they get inserted.
+    """
+
+    def __init__(self, item_result, sort_order):
```

Comment:
We should try to type-hint all new code as much as possible...
I appreciate the retroactively typing a large amount of existing code is difficult and time consuming - but all new code should have as much as is feasible, to prevent the existing delta of un-typed code increasing.
It assists in our ability to validate code via CI, and improves the robustness of the code base over time.
Also - we will in future be enforcing a policy of not allowing the percentage of an SDK that has typing to backslide.